src/semantics.rs

```diff
@@ -130,6 +130,14 @@ impl Validate for Expression {
                             _ => Err("In/NotIn operators only supports string/IP cidr operands".to_string())
                         }
                     },
+                    BinaryOperator::Contains => {
+                        match p.rhs {
+                            Value::String(_) | Value::IpCidr(_) => {
```

Comment:
We probably shouldn't support `IpCidr` in here as the LHS can only be a single IP address at the point, so `contains` actually makes little sense.

Reply:
fixed